### ISSUE TITLE
Godot doesn't always reload externally modified .tscn files

### ISSUE THREAD

**Issue by PZerua** [godotengine/godot#73602]
### Godot version

4.0 beta17, rc1, rc2

### System information

Windows 11, GTX 970

### Issue description

When a .tscn scene is opened in Godot and gets modified externally, Godot doesn't always detect the change and keeps displaying the old version. If you then save the scene from Godot, the new changes are discarded. Noticed this while doing a small game for the GGJ2023 a couple of weeks ago, we had many issues working with git because every once in a while, when doing git pull we ended with mangled versions of our scenes and resources. Probably we did a git pull and then run the game to check new changes, which saves all the scenes by default (but the old versions due to this issue).

While writting the issue I noticed that it always happens if you drag and drop a modified scene in the project folder (not inside Godot, the folder from your file explorer), but works well if the file is modified by an external editor or drag and dropped inside Godot itself.

It probably happens with other resources too, I need to check more in depth.

I guess a workaround would be always closing Godot before doing git pull.

### Steps to reproduce

- Load the MRP and open main.tscn inside Godot
- Copy main.tscn outside the project folder and modify this copied version from an external editor (for instance remove the line "size = Vector3(1, 2, 1)" from the CSGBox3D, which should give a default box size
- Copy the scene back to the project folder from your OS file explorer (tested in Windows)
- See how Godot doesn't detect the change
- Extra: Save the scene and see how the new change is lost

### Minimal reproduction project

[git_bug.zip](https://github.com/godotengine/godot/files/10778063/git_bug.zip)

**Comment by KoBeWi** [comment]
Can't reproduce.

https://user-images.githubusercontent.com/2223172/219984127-7de4c0bf-6715-4ea2-9af4-3379c056a2f6.mp4

Sounds like #61423, but that issue has been fixed.

**Comment by PZerua** [comment]
The issue always happens to me if I exactly follow the steps to reproduce. Remember you have to copy the scene outside the project, modify the copied scene and then move it back and overwrite (from os file explorer). I'll try to make a video later so it's clearer.

**Comment by KoBeWi** [comment]
I tried that do and the result was the same (which is not surprising, because there is no difference where you modify the file).

**Comment by PZerua** [comment]
Sorry you are right, I've been trying and I think I did something else. Instead of copying the scene I moved it, then saved from Godot (which generates a scene with the same name but with different uid). Then I modified the moved scene and dragged it back to the project folder.

[bug.webm](https://user-images.githubusercontent.com/8772233/220088646-3a7102d2-0c8f-414b-b045-683074a9cdad.webm)

It makes sense I had git issues because I remember the issue https://github.com/godotengine/godot/issues/62258 got fixed the day after we finished, so probably we were facing a mix of this two issues. My collegues had scenes and resources with different uids and Godot wasn't always reloading when we did git pull.
